breathe/renderer/sphinxrenderer.py

```diff
@@ -2369,11 +2370,22 @@ def visit_docdot(self, node) -> List[Node]:
     def visit_docdotfile(self, node) -> List[Node]:
         """Translate node from doxygen's dotfile command to sphinx's graphviz directive."""
         dotcode = ""
+        dot_file_path = node.name  # type: str
+        # Doxygen v1.9.3+ uses a relative path to specify the dot file.
+        # Previously, Doxygen used an absolute path.
+        # This relative path is with respect to the XML_OUTPUT path.
+        # furthermore, Doxygen v1.9.3+ will copy the dot file into the XML_OUTPUT
+        if not os.path.isabs(dot_file_path):
+            # use self.project_info.project_path as the XML_OUTPUT path,
+            # and make it absolute with consideration to the conf.py path
+            dot_file_path = os.path.abspath(
+                self.app.confdir + os.sep + self.project_info.project_path() + dot_file_path
```

Comment:
Is it any more preferable to use os.path.join here? I'm guessing not but I'm not sure why :)

Reply:
IDK. I didn't try it. I was more concerned with concatenating relative paths.

- `app.confdir` is an absolute path.
- `project_info.project_path()` might be absolute or relative. This may differ depending on how users configure [`breathe_projects`](https://breathe.readthedocs.io/en/latest/directives.html#confval-breathe_projects).
- `dot_file_path` is clearly compensated for absolute or relative paths (the point of this PR).

After giving this more thought, I should probably add a conditional statement for if `project_path()` yields an absolute path.

Reply:
I also didn't check if project_path() automatically ensures there is a trailing path delimiter (`os.sep`).

Reply:
Looking at the docs, I think [`os.path.abspath()`](https://docs.python.org/3/library/os.path.html#os.path.abspath) might be a better solution for cross-platform compatibility. [`os.path.join()`](https://docs.python.org/3/library/os.path.html#os.path.join) seems to have a different tactic when encountering absolute paths (specifically noted for Windows).